package-system/AWSNativeSDK/docker_build_aws_sdk.sh

```diff
@@ -6,85 +6,188 @@
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 #
 
-# Validate the src path
-src_path=src
-if [ ! -d $src_path ]
+
+# Validate the bld path input
+BUILD_FOLDER=${DOCKER_BUILD_PATH}
+if [ "${BUILD_FOLDER}" == "" ]
+then
+    echo "Missing required build target folder environment"
+    exit 1
+elif [ "${BUILD_FOLDER}" == "temp" ]
 then
-    echo "Missing src path"
+    echo "Build target folder environment cannot be 'temp'"
     exit 1
 fi
 
 
-# Make sure the build path is clear
-bld_path=build
-rm -rf $bld_path || (echo "Command: rm -rf $bld_path failed" ; exit 1)
-mkdir $bld_path
+# Locate the dependent OpenSSL package
+OPENSSL_REGEX='(OpenSSL-([A-Za-z0-9\.\-]*)-(linux|linux-aarch64))'
+[[ $DOWNLOADED_PACKAGE_FOLDERS =~ $OPENSSL_REGEX ]]
+DEPENDENT_OPENSSL=${BASH_REMATCH[1]}
 
+if [ $DEPENDENT_OPENSSL == "" ]
+then
+    echo "Unable to detect dependent OpenSSL package"
+    exit 1
+fi
+DEPENDENT_OPENSSL_BASE=$WORKSPACE/temp/${DEPENDENT_OPENSSL}/OpenSSL
+if [ ! -d ${DEPENDENT_OPENSSL_BASE} ]
+then
+    echo "Unable to detect dependent OpenSSL package at ${DEPENDENT_OPENSSL_BASE}"
+    exit 1
+fi
+echo "Detected dependent OpenSSL package at ${DEPENDENT_OPENSSL_BASE}"
+
+
+# Locate the dependent ZLIB package
+OPENZLIB_REGEX='(zlib-([A-Za-z0-9\.\-]*)-(linux|linux-aarch64))'
```

Comment:
Should this use the '+' to make sure that at least one character is matched between `zlib-` and `-(linux|linux-aarch64)` ?